EventListener/UserContextSubscriber.php

```diff
@@ -124,6 +126,15 @@ public function onKernelResponse(FilterResponseEvent $event)
         }
 
         $response = $event->getResponse();
+        $request = $event->getRequest();
+
+        // hash has changed, session has most certainly changed, prevent setting incorrect cache
+        if ($request->attributes->get($this->hashHeader) !== $request->headers->get($this->hashHeader)) {
```

Comment:
- agree with the break, this should be not happened if this->requestMatcher->matches. Will provide an updated version
- agree also with the too generic user context hash that can happened on resources like /translations, js, css. We prevent that in our varnish vcl with some return(hash) before the user context lookup. But i'm not sure what needs to be done on this topic.